inc/TRestRawDreamToSignalProcess.h

```diff
@@ -0,0 +1,198 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+#ifndef RestCore_TRestFEUDreamToSignalProcess
+#define RestCore_TRestFEUDreamToSignalProcess
+
+#include "TRestRawSignalEvent.h"
+#include "TRestRawToSignalProcess.h"
+
+#include <arpa/inet.h>
+#include <signal.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include <TError.h>
+#include <TF1.h>
+#include <TH1D.h>
+#include <TH2D.h>
+#include <TMath.h>
+#include <TROOT.h>
+#include <TSystem.h>
+#include <TTree.h>
+
+#include <arpa/inet.h>
+#include <iostream>
+#include <list>
+#include <map>
+#include <string>
+#include <vector>
+#include "math.h"
+
+//! An process to read binary data from FEUDream electronics
+class TRestRawDreamToSignalProcess : public TRestRawToSignalProcess {
+   protected:
+    class DataLineDream {
+       public:
+        DataLineDream() { data = 0; }
+        ~DataLineDream() {}
+        void ntohs_() { data = ntohs(data); };
+        bool is_final_trailer() const { return (((data)&0x7000) >> 12) == 7; }   // X111
+        bool is_end_of_event() const { return (((data)&0x7800) >> 11) == 0xF; }  // X1111
+        bool is_data_trailer() const { return (((data)&0x6000) >> 13) == 2; }    // X10X
+        bool is_first_line() const { return (((data)&0x7000) >> 12) == 3; }      // X011
+        bool is_data() const { return (((data)&0x7000) >> 12) == 0; }            // X000
+        bool is_data_zs() const { return (((data)&0x6000) >> 13) == 0; }         // X00X
+        bool is_channel_ID() const { return (((data)&0x7000) >> 12) == 1; }      // X001
+        bool is_Feu_header() const { return (((data)&0x7000) >> 12) == 6; }      // X110
+        bool is_data_header() const { return (((data)&0x6000) >> 13) == 1; }     // X01X
+        bool get_zs_mode() const { return (((data)&0x400) >> 10); }
+        int get_Feu_ID() const { return (((data)&0xFF)); }
+        long int get_finetstp() const { return (((data)&0x0007)); }
+        //#define GET_FINETSTP(word)      (word & 0x0007)
+        int get_sample_ID() const { return (((data)&0xFF8) >> 3); }
+        int get_channel_ID() const { return (((data)&0x3F)); }
+        int get_dream_ID() const { return (((data)&0xE00) >> 9); }  // non-zS mode
+        int get_dream_iD() const { return (((data)&0xE00) >> 6); }  // for zS mode
```

Comment:
The difference between upper and lower case in `get_dream_iD()` is error prone. I would suggest to do `get_dream_iD( bool zS)`